external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -136,3 +150,46 @@ pub(crate) fn write_prologue(
 
     Ok(())
 }
+
+pub fn update_compiler_toolchain(
+    file: &mut LockFile,
+    compiler_version: String,
+    build_config: &BuildConfig,
+) -> Result<()> {
+    let mut toml_string = String::new();
+    file.read_to_string(&mut toml_string)?;
+    file.seek(SeekFrom::Start(0))?;
+    let mut toml = toml_string.parse::<toml_edit::Document>()?;
+    let move_table = toml["move"].as_table_mut().ok_or(std::fmt::Error)?;
+    move_table["compiler-version"] = toml_edit::value(compiler_version);
+    let compiler_flags = toml::to_string(build_config)?.parse()?;
+    let compiler_flags = to_toml_edit_value(&compiler_flags);
```

Comment:
I investigated many many things before finally resorting to this approach so that we can serialize `BuildConfig` without manual logic (that way, changes to `BuildConfig` are included in `Move.lock` files). Approach goes like this:

- serialize `BuildConfig`, then parse it so we get it as a `toml` value
- use function to convert `BuildConfig` `toml` values to `toml_edit` values so we can update the toml value with the type(s) that please `toml_edit`

Reply:
This is a bit hacky, so let me know your thoughts on it (and feel free to say it's a bad idea) but you could conceivably take the existing lock file and concatenate the build config information.  When there are existing build configs, you could update them by reading the existing lock file, removing the build config with `toml_edit` and then proceeding to concatenate the new value.

Reply:
Yeah, the thing that bothers me is concatting to the end of file we we already have some dependency information before this info (it gets committed to the lock when it succeeds, and I hesitate whether we want to update the compiler toolchain versioning in that part of the logic).

I really want there to be a reasonable ordering in the `Move.lock` file, so putting the toolchain info under `[move]` right at the top feels better than concatting--we'll see this info at a glance when opening the file, not after it's potentially buried (or shifted around) when the lock file schema grows.